packages/suite/src/actions/wallet/cardanoStakingActions.ts

```diff
@@ -112,24 +123,41 @@ export const fetchTrezorPools = (network: 'ADA' | 'tADA') => async (dispatch: Di
     });
 
     // Fetch ID of Trezor stake pool that will be used in delegation transaction
-    const url =
+    const url_pools =
         cardanoNetwork === 'mainnet'
             ? CARDANO_STAKE_POOL_MAINNET_URL
             : CARDANO_STAKE_POOL_PREVIEW_URL;
 
-    try {
-        const response = await fetch(url, { credentials: 'same-origin' });
-        const responseJson = await response.json();
+    // Fetch dreps for transaction withdrawal
+    const url_dreps = cardanoNetwork === 'mainnet' ? CARDANO_MAINNET_DREPS : CARDANO_PREVIEW_DREPS;
 
-        if (!responseJson || !('next' in responseJson) || !('pools' in responseJson)) {
+    try {
+        const responsePools = await fetch(url_pools, { credentials: 'same-origin' });
+        const responsePoolsJson = await responsePools.json();
+
+        if (
+            !responsePoolsJson ||
+            !('next' in responsePoolsJson) ||
+            !('pools' in responsePoolsJson)
+        ) {
             // todo: even if this happens, error will be overridden by this bug
             // https://github.com/trezor/trezor-suite/issues/5485
             throw new Error('Cardano: fetchTrezorPools: Invalid data format');
         }
 
+        const responseDreps = await fetch(url_dreps, { credentials: 'same-origin' });
+        const responseDrepsJson = await responseDreps.json();
+
+        if (!responseDrepsJson || !('drep' in responseDrepsJson)) {
+            // todo: even if this happens, error will be overridden by this bug
+            // https://github.com/trezor/trezor-suite/issues/5485
+            throw new Error('Cardano: fetchTrezorDreps: Invalid data format');
+        }
+
         dispatch({
-            type: CARDANO_STAKING.SET_TREZOR_POOLS,
-            trezorPools: responseJson as PoolsResponse,
+            type: CARDANO_STAKING.SET_TREZOR_DATA,
+            trezorPools: responsePoolsJson as PoolsResponse,
+            trezorDreps: responseDrepsJson as DrepsResponse,
```

Comment:
there can be only one trezor drep
```suggestion
            trezorDrep: responseDrepJson as DrepResponse,
```